jina/serve/runtimes/gateway/grpc/gateway.py

```diff
@@ -18,26 +19,30 @@ class GRPCGateway(BaseGateway):
 
     def __init__(
         self,
+        args: 'argparse.Namespace',
         port: Optional[int] = None,
         grpc_server_options: Optional[dict] = None,
         ssl_keyfile: Optional[str] = None,
         ssl_certfile: Optional[str] = None,
         **kwargs,
     ):
         """Initialize the gateway
+        :param args: runtime args
         :param port: The port of the Gateway, which the client should connect to.
         :param grpc_server_options: Dictionary of kwargs arguments that will be passed to the grpc server as options when starting the server, example : {'grpc.max_send_message_length': -1}
         :param ssl_keyfile: the path to the key file
         :param ssl_certfile: the path to the certificate file
         :param kwargs: keyword args
         """
-        super().__init__(**kwargs)
+        super().__init__(args=args, **kwargs)
+        self._setup_instrumentation()
```

Comment:
this one expects an args in self. However I don't think it's possible. What should be done instead is one of these options:
* extract all needed parameters from args and accept them in signature (and possible turn the mixin into a set of helper functions)
* modify the usage of _setup_instrumentation in order to follow monitoring: it is to be applied on runtimes not on gateway class and the runtime receives instrumentation objects and injects them into the gateway class (we already do this dependency injection logic in set_streamer method)